### PR TITLE
Update ETCD XCCDF and OVAL for ocp4-isms

### PR DESCRIPTION
**Pull request by redhatrises**
- Add runtime checks
- Update ETCD XCCDF and OVAL for ocp4-isms

### CODE REVIEW

File: applications/openshift/etcd/etcd_peer_cert_file/rule.yml [hunk 1]
```diff
@@ -9,7 +9,11 @@ description: |-
     edit the <tt>etcd</tt> configuration file
     <tt>/etc/etcd/etcd.conf</tt> on the master and adding a certificate
     to <tt>ETCD_PEER_CERT_FILE</tt>:
+{{%- if product == "ocp4" %}}
+    <pre>ETCD_PEER_CERT_FILE=/etc/ssl/etcd/system:etcd-peer:<i><etcd_dns_name</i>.crt</pre>
```
Comment (shawndwells) [inline applications/openshift/etcd/etcd_peer_cert_file/rule.yml]:
```suggestion
    <pre>ETCD_PEER_CERT_FILE=/etc/ssl/etcd/system:etcd-peer:<i><etcd_dns_name></i>.crt</pre>
```

File: applications/openshift/etcd/etcd_cert_file/rule.yml [hunk 2]
```diff
@@ -9,7 +9,11 @@ description: |-
     edit the <tt>etcd</tt> configuration file
     <tt>/etc/etcd/etcd.conf</tt> on the master and adding a certificate
     to <tt>ETCD_CERT_FILE</tt>:
+{{%- if product == "ocp4" %}}
+    <pre>ETCD_CERT_FILE=/etc/ssl/etcd/system:etcd-server:<i><etcd_dns_name</i>.crt</pre>
```
Comment (shawndwells) [inline applications/openshift/etcd/etcd_cert_file/rule.yml]:
```suggestion
    <pre>ETCD_CERT_FILE=/etc/ssl/etcd/system:etcd-server:<i><etcd_dns_name></i>.crt</pre>
```

File: applications/openshift/etcd/etcd_key_file/oval/shared.xml [hunk 3]
```diff
@@ -27,7 +28,13 @@
 
   <ind:textfilecontent54_state id="state_etcd_key_file"
   version="1">
-    <ind:subexpression datatype="string" operation="pattern match">^/etc/etcd/server.key$</ind:subexpression>
+    <ind:subexpression datatype="string" operation="pattern match">^/etc/etcd/server.key|/etc/ssl/etcd/system\:etcd-.*\:etcd.*\.key$</ind:subexpression>
   </ind:textfilecontent54_state>
 
+{{%- if product == "ocp4" %}}
+  {{{ oval_ocp_service_runtime_config(command="etcd", option="--key-file", value="/etc/ssl/etcd/system\:etcd-server\:etcd.*\.key", option_id="key_file") }}}
```
Comment (shawndwells) [inline applications/openshift/etcd/etcd_key_file/oval/shared.xml]:
```suggestion
  {{{ oval_ocp_service_runtime_config(command="etcd", option="--key-file", value="/etc/ssl/etcd/system\:etcd-server\:etcd-.*\.key", option_id="key_file") }}}
```

Comment (shawndwells) [reply applications/openshift/etcd/etcd_key_file/oval/shared.xml]: to align with the ``ind:subexpression``

File: applications/openshift/etcd/etcd_peer_key_file/oval/shared.xml [hunk 4]
```diff
@@ -30,4 +31,10 @@
     <ind:subexpression datatype="string" operation="pattern match">^/etc/etcd/peer.key$</ind:subexpression>
   </ind:textfilecontent54_state>
 
+{{%- if product == "ocp4" %}}
+  {{{ oval_ocp_service_runtime_config(command="etcd", option="--peer-key-file", value="/etc/ssl/etcd/system\:etcd-peer\:etcd.*\.key", option_id="peer_key_file") }}}
```
Comment (shawndwells) [inline applications/openshift/etcd/etcd_peer_key_file/oval/shared.xml]: XCCDF has:
``ETCD_PEER_KEY_FILE=/etc/ssl/etcd/system:etcd-peer:<i><etcd_dns_name</i>.key``

But OVAL has:
``value="/etc/ssl/etcd/system\:etcd-peer\:etcd.*\.key"``

Which is correct?

Comment (shawndwells) [reply applications/openshift/etcd/etcd_peer_key_file/oval/shared.xml]: Or is there a naming convention that etcd dns names will have ``etcd*``?

File: applications/openshift/etcd/etcd_peer_key_file/rule.yml [hunk 5]
```diff
@@ -9,7 +9,11 @@ description: |-
     edit the <tt>etcd</tt> configuration file
     <tt>/etc/etcd/etcd.conf</tt> on the master on the master and
     adding a key file to <tt>ETCD_PEER_KEY_FILE</tt>:
+{{%- if product == "ocp4" %}}
+    <pre>ETCD_PEER_KEY_FILE=/etc/ssl/etcd/system:etcd-peer:<i><etcd_dns_name</i>.key</pre>
```
Comment (shawndwells) [inline applications/openshift/etcd/etcd_peer_key_file/rule.yml]:
```suggestion
    <pre>ETCD_PEER_KEY_FILE=/etc/ssl/etcd/system:etcd-peer:<i><etcd_dns_name></i>.key</pre>
```

File: applications/openshift/etcd/etcd_unique_ca/oval/shared.xml [hunk 6]
```diff
@@ -24,7 +24,13 @@
   </ind:textfilecontent54_object>
 
   <ind:textfilecontent54_state id="state_etcd_unique_ca" version="1">
-    <ind:subexpression datatype="string" operation="pattern match">^/etc/etcd/ca.crt$</ind:subexpression>
+    <ind:subexpression datatype="string" operation="pattern match">^/etc(|/ssl)/etcd/ca.crt$</ind:subexpression>
```
Comment (shawndwells) [inline applications/openshift/etcd/etcd_unique_ca/oval/shared.xml]: If OCP3 is being evaluated, is ``/etc/ssl/etcd/ca.crt`` acceptable, or does OCP3 expect ``/etc/etcd/ca.crt``?

Comment (redhatrises) [reply applications/openshift/etcd/etcd_unique_ca/oval/shared.xml]: OCP3 sets up the CA as `/etc/etcd/ca.crt`, so changing the path and such I don't think makes sense.

File: applications/openshift/etcd/etcd_peer_key_file/oval/shared.xml (hunk 4 above)
Comment (redhatrises) [reply applications/openshift/etcd/etcd_peer_key_file/oval/shared.xml]: @shawndwells I don't think that you can guarantee an FQDN is going to be the same at all.

Comment (shawndwells) [reply applications/openshift/etcd/etcd_peer_key_file/oval/shared.xml]: @redhatrises right, so shouldn't the oval be:

``value="/etc/ssl/etcd/system\:etcd-peer\:.*\.key"``

vs the current 
``value="/etc/ssl/etcd/system\:etcd-peer\:etcd.*\.key"``

Comment (redhatrises) [reply applications/openshift/etcd/etcd_peer_key_file/oval/shared.xml]: Good point. I see what you are saying now. Fixed.

**Pull request merged**